### ISSUE TITLE
PHP Fatal error:  Class GuzzleHttp\Client contains 1 abstract method

### ISSUE THREAD

**Issue by tussion** [guzzle/guzzle#992]
Encountered with following error.

PHP Fatal error:  Class GuzzleHttp\Client contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (GuzzleHttp\Event\HasEmitterInterface::getEmitter) in /Applications/MAMP/htdocs/guzzle/vendor/guzzlehttp/guzzle/src/Client.php on line 403

**Comment by mtdowling** [comment]
That's strange. Are you sure you have the right version of everything installed?

**Comment by mtdowling** [comment]
Closing for now. This seems like an environment specific issue.

**Comment by tussion** [comment]
Thanks for revert. Yes you are right, it was environment specific issue. Fixed now. 

Not sure it's right place to discuss but I would like say i really pissed of with PHP for asynchronous calls even with guzzle too. I know asynchronous calls & multi threading type things are not aval in PHP by default but i was sure that there must be some workaround for same.

In one of my project i have to update key value pairs over HTTP from one server to another server using backlink IPs. We encountered with many issues like backlink choked, Open file limits and throughput (around 100 Res/sec). We initially started with guzzle then we switched with forked curl process with "> /dev/null 2>&1 &"  to ensure that process gets properly forked and doesn’t log anywhere.
It works okis way except we lost control over codebase as well as debugging & logging. Then we tried  fsockopen but found it's blocking, it blocks until the socket is fully connected before returning it. Here is proof for same.

https://github.com/php/php-src/blob/master/main/streams/transports.c#L143
https://github.com/php/php-src/blob/master/main/network.c#L393
http://en.wikipedia.org/wiki/Transmission_Control_Protocol#Connection_establishment

Things were going more slower when SSL introduced ( https) .. https://github.com/php/php-src/blob/master/ext/openssl/xp_ssl.c#L434

finally we found some better approach is to open a persistent socket using pfsockopen. This will re-use earlier socket connections made by the PHP process, which doesn’t require a TCP handshake each time.
I would like to hear your comments here. Should I use any messaging solution like RabbitMQ or NodeJS based ( We already using StatsD so nodeJS is aval on production server)  so that i can control over HTTP Requests with proper logging & debugging and can get better throughput. Few people in Team are not willing to allow to use RabbitMQ etc because they are denying  messaging solution as it's against architecture design. 

Let me know what you suggest here.
